src/processing/action.py

```diff
@@ -1,45 +1,103 @@
 import torch
-from state import GameState, Frame, PlayerFrame, Keypoint
-from pose_estimation.pose_estimate import KeyPointNames, AngleNames
-
-COMBINATIONS = AngleNames.combinations
-ANGLE_NAMES = AngleNames.list
+from state import GameState, PlayerFrame, BallFrame, Box
+from args import DARGS
 
 class ActionRecognition:
-    def __init__(self, state: GameState):
+    def __init__(self, state: GameState, args=DARGS) -> None:
         self.state = state
+        self.THRESHOLD = args["shot_threshold"]
+        self.ANGLE_THRESHOLD = args["angle_threshold"]
 
-    @staticmethod
-    def is_shot(player_frame: PlayerFrame):
+    def pose_shot(self, player_frame: PlayerFrame, ANGLE_THRESHOLD):
         """
-        Takes in keypoint and angle data for a player in a frame and returns whether
-        or not the player is shooting. Currently uses a simple threshold heuristic.
+        Takes in keypoint and angle data for a player in a frame and returns the
+        weighted sum of the pose features. Currently uses a simple threshold
+        heuristic.
         """
+        keypoints = ["left_wrist", "right_wrist", "left_shoulder", "right_shoulder"]
+        for keypoint in keypoints:
```

Comment:
Right now is that NaNs for keypoints are given coordinates (-1,-1), and they will appear in the PlayerFrame.keypoints even if they are NaN